pkg/baremetal/v1alpha3/register.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +k8s:deepcopy-gen=false
+type BareMetalProviderSpecCodec struct {
```

Comment:
Please remove this. It seems unused, and anyways there will/should be "better" ways these days to encode/decode these CRDs.